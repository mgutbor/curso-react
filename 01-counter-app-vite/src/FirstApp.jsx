import PropTypes from 'prop-types';

export const FirstApp = ({ title, subTitle, name }) => {

  return (
    <>
      <h1>{ title }</h1>
      <p>{ subTitle }</p>
      <p>{ name }</p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.number.isRequired
}

FirstApp.defaultProps = {
  subTitle: 'No hay subtitulo',
  name: 'Manuel'
}