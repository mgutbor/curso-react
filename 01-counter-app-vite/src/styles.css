html, body{
    background-color: #cacaca;
    color: red;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 1rem;
}

button{
    padding: 15px;
    font-size: 1.3rem;
}