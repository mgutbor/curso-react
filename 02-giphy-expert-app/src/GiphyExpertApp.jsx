import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GiphyExpertApp = () => {

  const [categories, setCategories] = useState( ['Pokemon'] );

  const onAddCategory = (newCategory) => {
    if(categories.findIndex( item =>  newCategory.toLowerCase() === item.toLowerCase()) !== -1 ) return;
    // if (categories.includes(newCategory)) return;
    setCategories([newCategory, ...categories]);
  }

  return (
    <>
      <h1>GiphyExpertApp</h1>

      <AddCategory onNewCategory={ onAddCategory } />

      { 
        categories.map( (category) => (
          <GifGrid 
            key={category} 
            category={category}
          />
        ))
      }
    </>
  )
}
