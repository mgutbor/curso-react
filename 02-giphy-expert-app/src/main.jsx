import React from 'react';
import ReactDOM from 'react-dom/client';
import {GiphyExpertApp} from './GiphyExpertApp';
import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <GiphyExpertApp />
  </React.StrictMode>,
)
