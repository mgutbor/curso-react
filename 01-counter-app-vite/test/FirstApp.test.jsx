import { render } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Pruebas sobre el componente FirstApp', () => {

    // test('debe hacer match con el snapshot', async() => {

    //     const title = "Hola, soy una prueba";
    //     const subtitulo = 123;
    //     const { container } = render( <FirstApp title={title} subTitle={subtitulo}/> )

    //     expect(container).toMatchSnapshot(); 
    // });

    test('debe mostrar el titulo en un h1', async() => {

        const title = "Hola, soy una prueba";
        const { getByText, getByTestId } = render( <FirstApp title={title} /> )

        expect(getByText(title)).toBeTruthy();

        expect(getByTestId('test-title').innerHTML).toContain(title);
    })

    test('debe mostrar el subtitulo enviado por props', async() => {

        const title = "Hola, soy una prueba";
        const subTitle = 'Subtitulo de la app';
        const { getByText } = render(
            <FirstApp 
                title={title}
                subTitle={subTitle}
            />
        );

        expect(getByText(subTitle)).toBeTruthy();
    })
});