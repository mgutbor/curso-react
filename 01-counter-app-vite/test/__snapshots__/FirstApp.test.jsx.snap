// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente FirstApp debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola, soy una prueba
  </h1>
  <p>
    123
  </p>
  <p>
    Manuel
  </p>
</div>
`;
